tests/integrationv2/test_buffered_send.py

```diff
@@ -0,0 +1,123 @@
+    port = next(available_ports)
+
+    random_bytes = data_bytes(SEND_DATA_SIZE)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        cipher=cipher,
+        data_to_send=random_bytes,
+        insecure=True,
+        protocol=protocol)
+
+    if other_provider is S2N:
+        client_options.extra_flags = ['--buffered-send', buffer_size]
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        data_to_send=None,
+        insecure=True,
+        protocol=protocol)
+ 
+    server_options.extra_flags = ['--buffered-send', buffer_size]
+    if fragment_pref is not None:
+        server_options.extra_flags.append(fragment_pref)
+
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.cipher = None
+
+    server = managed_process(provider, server_options, timeout=5)
+    client = managed_process(other_provider, client_options, timeout=5)
+
+    for results in client.get_results():
+        results.assert_success()
+
+    for results in server.get_results():
+        results.assert_success()
+        if provider is S2N: random_bytes in results.stdout
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("other_provider", [S2N], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", PROTOCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("buffer_size", SEND_BUFFER_SIZES, ids=get_parameter_name)
+@pytest.mark.parametrize("fragment_pref", [None, "--prefer-low-latency", "--prefer-throughput"], ids=get_parameter_name)
+def test_s2n_buffered_send_client(managed_process, cipher, other_provider, provider, protocol, certificate, buffer_size, fragment_pref):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(SEND_DATA_SIZE)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        cipher=cipher,
+        data_to_send=random_bytes,
+        insecure=True,
+        protocol=protocol)
+
+    client_options.extra_flags = ['--buffered-send', buffer_size]
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        data_to_send=None,
+        insecure=True,
+        protocol=protocol)
+
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.cipher = None
+
+    server = managed_process(provider, server_options, timeout=5)
+    client = managed_process(other_provider, client_options, timeout=5)
+
+    for results in client.get_results():
+        results.assert_success()
+
+    for results in server.get_results():
+        results.assert_success()
+
```

Comment:
I was relying on the exit code, since OpenSSL doesn't print the data in a a convenient way.

-- But I've check other implementations and added checks to ensure that we have a connection. And don't have any output on stderr.

Reply:
OpenSSL should print any data it receives to stdout. How is it behaving instead?

Reply:
It is printing data to STDOUT but in a highly non-contiguous way